File: cpp/Problem268.cpp
/*
 * Problem: 268. Missing Number
 * URL: https://leetcode.com/problems/missing-number/
 * Difficulty: Easy
 * Tags: Bitwise Operation
 */

#include <vector>

using namespace std;

class Solution {
public:
	int missingNumber(vector<int>& nums) {
		int m = 0;
		for (int i = 1, size = nums.size(); i <= size; i++) {
			m ^= i;
		}
		for (int num : nums) {
			m ^= num;
		}
		return m;
	}
};

File: python/Problem268.py
# Problem: 268. Missing Number
# URL: https://leetcode.com/problems/missing-number/
# Difficulty: Easy
# Tags: Bitwise Operation

from typing import List

class Solution:
  def missingNumber(self, nums: List[int]) -> int:
    m: int = 0
    for i in range(1, len(nums) + 1):
      m ^= i
    for num in nums:
      m ^= num
    return m